src/libs/ajax/DataRepo.ts

```diff
@@ -81,6 +80,60 @@ export type SnapshotBuilderSettings = {
   datasetConceptSets?: PrepackagedConceptSet[];
 };
 
+export interface Criteria {
+  // This is the ID for either the domain or the program data option
+  id: number;
+  kind: SnapshotBuilderOptionTypeNames;
+  count?: number;
+}
+
+export interface DomainCriteria extends Criteria {
```

Comment:
are these all still api types, just moved to a new file? I find it confusing to tell the difference between the UI types and API types if they have the same name (instead of for example DomainCriteriaApi). I like that you moved them over to this file, but I also like the distinct Api names. What are your thoughts on this?

Reply:
You're correct, these are all api types moved to a new file. There are differences with the contents of the UI and API types: 

For example, `CriteriaGroup` looks completely similar here: 

<img width="1068" alt="Screenshot 2024-04-26 at 2 34 46 PM" src="https://github.com/DataBiosphere/terra-ui/assets/63474660/392e585a-0cd0-4760-879b-6e77ad123f08">

But, as we beginning climbing up to`AnyCriteria`, and then `DomainCriteria` we can see that they have different specifications: 
<img width="1066" alt="Screenshot 2024-04-26 at 2 35 40 PM" src="https://github.com/DataBiosphere/terra-ui/assets/63474660/c7f813b6-a7e4-4b26-b4ac-bd48795d0687">


I agree with you that including the `Api` is helpful, that's why whenever they're used outside of `DataRepo` we use an alias. ex: 

`import { DomainCriteria as DomainCriteriaApi } from 'DataRepo'`

Reply:
I put the API and UI types in the same order that way, it's a lot easier to compare, I found it really difficult to tell the differences when they were in one file. Does this make sense?

Reply:
Okay, thanks for your explanation! I guess my suggestion would be to not just rely on aliasing but to have the names actually be different when the types are different. Or maybe i just need to understand, what was the reason in removing the 'Api' from the api types when they differ from the UI types?

Reply:
Removing the `Api` was because these types are now in `DataRepo` which I think suggests that they are API types. Having `Api` in the type name when they were in `DatasetBuilderUtils` was useful because we could differentiate them but since they're in two difference places, I personally don't think they're needed. What do you think?

Reply:
Okay, sounds good. If you and others don't find it confusing then its fine with me!

Reply:
Ok sweet. When I mean bias, I also mean the bias of writing this code. The a

Reply:
Did we get a chance to discuss this further?

Reply:
I think some of the confusion goes away if we are consistent with what types go where, and what names we use. If the ones in `DataRepo.ts` always match the API names and type definitions, and the ones in `DatasetBuilderUtils.ts` are our custom types, then we can rely on that to tell the difference. And then only alias when it doesn't add more confusion.